src/Workspaces/CSharp/Portable/Simplification/Reducers/CSharpMiscellaneousReducer.cs

```diff
@@ -110,14 +111,61 @@ public CSharpMiscellaneousReducer() : base(s_pool)
             OptionSet optionSet,
             CancellationToken cancellationToken)
         {
-            if (node.Statements.Count == 1 &&
-                CanHaveEmbeddedStatement(node.Parent) &&
-                !optionSet.GetOption(CSharpCodeStyleOptions.PreferBraces).Value)
+            if (node.Statements.Count != 1)
             {
-                return node.Statements[0];
+                return node;
             }
 
-            return node;
+            if (!CanHaveEmbeddedStatement(node.Parent))
+            {
+                return node;
+            }
+
+            switch (optionSet.GetOption(CSharpCodeStyleOptions.PreferBraces).Value)
+            {
+                case PreferBracesPreference.Always:
+                    return node;
+
+                case PreferBracesPreference.WhenMultiline:
+                    // Braces are optional in several scenarios for 'when_multiline', but are only automatically removed
+                    // in a subset of cases where all of the following are met:
+                    //
+                    // 1. This is an 'if' statement
+                    // 1. The 'if' statement does not have an 'else' clause and is not part of a larger 'if'/'else if'/'else' sequence
+                    // 2. The 'if' statement is not considered multiline
```

Comment:
what's the intution here between these rules.

Reply:
The new rules are meant to support the .NET Foundation recommended code style for braces. While braces are allowed to be omitted in several places, it does not specifically encourage their removal in these places. I implemented the reducer to take a conservative approach here that stays in the bounds of what is allowed and was also easy to implement.

Reply:
Or, in other words... why is this not just using/referring to the same code from the analyzer. 

My gut is telling me because the way braces work is that we never really try to aggressively remove.  instead, it's more about what we *allow* the user to leave off, but it's always ok to have the braces.

So it's a bit unclear to me why we necessarily need this code.  Could you clarify?  Thanks!

Reply:
ok.  i see... 

I have mixed feelings here.  Primarily because the option is a preference about when braces are forced, but not really an affirmative statement about when they should be more aggressively removed.

However, i can see that we inferred 'never' to have that meaning.  So in that sense, this fits that approach.  This is also not likely to be a problem for people since it is fairly conservative.

Reply:
We can revisit this if it poses a problem in the future.

Reply:
Sounds good to me.